### DISCUSSION TITLE
SOAP Attachments

### DISCUSSION THREAD

**Discussion by d-wieser** [quarkiverse/quarkus-cxf#361] [Q&A] [answered]
Hello, 

first of all - thanks for providing this extension!

I´d like to implements a Contract First WebService. The code generation works well, but the build fails with the following exception:

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:2.6.3.Final:build (default) on project <xxx-service>: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkiverse.cxf.deployment.QuarkusCxfProcessor#build threw an exception: org.apache.ws.commons.schema.XmlSchemaException: Unable to locate imported document at 'http://ws-i.org/profiles/basic/1.1/swaref.xsd', relative to 'schema2.xsd'. 
```

I figured out that it is caused by an @XmlAttachmentRef Annotation at a DataHandler Attribute for an attachment.

```
    // generated code   
    @XmlElement(required = true)
    @XmlAttachmentRef // if removed, build works fine
    @XmlSchemaType(name = "anyURI")
    protected DataHandler attachment;
```

I there a good workaround without breaking the Contract? 

setup:
quarkus-cxf  0.13.01
cxf-codegen-plugin 3.5.0

**Comment by shumonsharif** [comment]
Hi @jwieser Yes, a workaround should certainly be doable with bindings, combined with an extension such as `jaxb2-annotate-plugin`.  Please attach a simple reproducer in case you need further help!

Having said that, this is probably also an issue in the extension - will try to have a look.

**Comment by shumonsharif** [comment]
I'm actually unable to recreate the issue you're having using a contract first SwA WSDL ... at this stage, without a reproducer, it's hard to say why you may be seeing the exception.  My best guess is that there may be an issue with your schemas or WSDL itself.

**Comment by d-wieser** [comment, marked as answer]
Tried a Code First approach by adding the folloing attribute to the Fruit Class from the Example in the Docs -  does not build. Just added the code to the integration-tests of the Extension.  I hope it is not a issue just on my machine or setup.

```
    @XmlAttachmentRef
    private DataHandler attachment;
```

**Reply by shumonsharif** [reply]
@jwieser I tried that as well, build worked fine ... unfortunately, as mentioned previously, without a reproducer it'll be difficult to say for sure - and anything I suggest would be somewhat of a guess at this stage.

**Reply by d-wieser** [reply]
Well, seems to be a problem on my machine at work. Runs without any problem an my private one (unfortunately).  
Thank´s a lot.

**Reply by shumonsharif** [reply]
Given the issue is limited to your work laptop, did you verify that you can access the `http://ws-i.org/profiles/basic/1.1/swaref.xsd` schema URL?  If it's an accessibility issue, you could always try to use a `META-INF/jax-ws-catalog.xml` file to redirect to the CXF core bundled file.  Something like this may possibly work:

```xml
<catalog xmlns="urn:oasis:names:tc:entity:xmlns:xml:catalog" prefer="system">
      <rewriteSystem systemIdStartString="http://ws-i.org/profiles/basic/1.1"
            rewritePrefix="classpath:/schemas/wsdl"/>
</catalog>
```

**Reply by famod** [reply]
Smells like a proxy (or VPN) problem.

**Reply by d-wieser** [reply]
You made my day. That´s it, thanks,. Already had the idea to use a catalog file, but i did it wrong.
